modules/network/traffic-manager/main.bicep

```diff
@@ -89,6 +142,19 @@ resource trafficManagerEndpoints 'Microsoft.Network/TrafficManagerProfiles/Exter
   }
 }]
 
+resource trafficManagerdiagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(diagnosticWorkspaceId) || !empty(diagnosticEventHubAuthorizationRuleId) || !empty(diagnosticEventHubName) && !empty(diagnosticsLogs)) {
```

Comment:
this if statement will always be true because 'diagnosticsMetrics' will never be empty.

Reply:
```suggestion
resource trafficManagerdiagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if(enablediagnostics) {
```

Reply:
how about just adding a simple toggle.